rskj-core/src/test/java/co/rsk/peg/federation/FederationStorageProviderImplTests.java

```diff
@@ -979,6 +981,110 @@ void getNextFederationCreationBlockHeight_afterRSKIP186AndNoValueInStorage_retur
+        System.out.println(Arrays.toString(expectedScript.getProgram()));
+        byte[] serializedScript = BridgeSerializationUtils.serializeScript(expectedScript);
+        // Putting some value in the storage just to then assert that before fork, the storage won't be accessed.
+        storageAccessor.saveToRepository(LAST_RETIRED_FEDERATION_P2SH_SCRIPT_KEY.getKey(), serializedScript);
+
+        // Act
+
+        FederationStorageProvider federationStorageProvider = new FederationStorageProviderImpl(storageAccessor);
+
+        // Assert
+        assertEquals(Optional.empty(), federationStorageProvider.getLastRetiredFederationP2SHScript(activations));
+
+    }
+
+    @Test
+    void getLastRetiredFederationP2SHScript_afterRSKIP186_getsValueFromStorage() {
+
+        // Arrange
+
+        ActivationConfig.ForBlock activations = mock(ActivationConfig.ForBlock.class);
+        when(activations.isActive(ConsensusRule.RSKIP186)).thenReturn(true);
+
+        StorageAccessor storageAccessor = new InMemoryStorage();
+        Script expectedScript = new P2shErpFederationBuilder().build().getP2SHScript();
+
+        byte[] serializedScript = BridgeSerializationUtils.serializeScript(expectedScript);
+
+        storageAccessor.saveToRepository(LAST_RETIRED_FEDERATION_P2SH_SCRIPT_KEY.getKey(), serializedScript);
+
+        // Act
+
+        FederationStorageProvider federationStorageProvider = new FederationStorageProviderImpl(storageAccessor);
+        Optional<Script> actualValueOptional = federationStorageProvider.getLastRetiredFederationP2SHScript(activations);
+
+        // Assert
+
+        assertTrue(actualValueOptional.isPresent());
+        assertEquals(expectedScript, actualValueOptional.get());
+
+        // Setting in storage a different value to assert that calling the method again should return cached value
+
+        storageAccessor.saveToRepository(LAST_RETIRED_FEDERATION_P2SH_SCRIPT_KEY.getKey(), expectedScript.getProgram());
+
+        Optional<Script> actualCachedValue = federationStorageProvider.getLastRetiredFederationP2SHScript(activations);
+
+        assertTrue(actualCachedValue.isPresent());
+        assertEquals(expectedScript, actualCachedValue.get());
+
+    }
+
+    @Test
+    void getLastRetiredFederationP2SHScript_afterRSKIP186AndNoValueInStorage_returnsEmpty() {
+
+        // Arrange
+
+        ActivationConfig.ForBlock activations = mock(ActivationConfig.ForBlock.class);
+        when(activations.isActive(ConsensusRule.RSKIP186)).thenReturn(true);
+
+        StorageAccessor storageAccessor = new InMemoryStorage();
+        storageAccessor.saveToRepository(LAST_RETIRED_FEDERATION_P2SH_SCRIPT_KEY.getKey(), null);
+
+        // Act
+
+        FederationStorageProvider federationStorageProvider = new FederationStorageProviderImpl(storageAccessor);
+        Optional<Script> actualValueOptional = federationStorageProvider.getLastRetiredFederationP2SHScript(activations);
```

Comment:
the `Optional` suffix does not add much value

Reply:
Updated.